flaky test executor/stale_txn "database doesn't exist"

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

<!-- a step by step guide for reproducing the bug. -->

https://do.pingcap.net/jenkins/blue/organizations/jenkins/pingcap%2Ftidb%2Fghpr_check/detail/ghpr_check/37137/pipeline

### 2. What did you expect to see? (Required)

### 3. What did you see instead (Required)

```
time="2025-08-20T01:53:48+08:00" level=error msg="1 tests failed\n"

time="2025-08-20T01:53:48+08:00" level=error msg="run test [executor/stale_txn] err: sql:select * from information_schema.ddl_jobs as of timestamp now();: run \"select * from information_schema.ddl_jobs as of timestamp now();\" at line 4 err Error 1049 (42000): database doesn't exist"

make: *** [Makefile:170: integrationtest] Error 1
```

### 4. What is your TiDB version? (Required)

<!-- Paste the output of SELECT tidb_version() -->

--- a/tests/integrationtest/r/executor/stale_txn.result
+++ b/tests/integrationtest/r/executor/stale_txn.result
@@ -1,4 +1,4 @@
-select * from information_schema.ddl_jobs as of timestamp now(6) - interval 1 second;
+select * from information_schema.ddl_jobs as of timestamp now(6) - interval 1 microsecond;
 drop table if exists t1;
 create table t1 (id int primary key, v int);
 insert into t1 values(1, 10);

--- a/tests/integrationtest/t/executor/stale_txn.test
+++ b/tests/integrationtest/t/executor/stale_txn.test
@@ -1,7 +1,7 @@
 # TestIssue35686
 ## This query should not panic
 --disable_result_log
-select * from information_schema.ddl_jobs as of timestamp now(6) - interval 1 second;
+select * from information_schema.ddl_jobs as of timestamp now(6) - interval 1 microsecond;
 --enable_result_log
 
 # TestIssue31954

